File: src - java/Esercizi/SpostaTestaInCoda/SpostaTestaInCoda.java
	private Record getPuntCoda() {
		if (first == null) // se la coda è vuota...
			return null; // ...restituisce null

		Record p = first;
		while (p.succ != null)
			// altrimenti cerca la coda
			p = p.succ;

		return p; // e la restituisce
	}
	
	public boolean spostaTestaInCoda() {
		Record p = getPuntCoda(); // restituisce il punt. alla
									// coda se c'è
		if (p != null) {
			p.succ = first; // il successivo della coda è first
			first = first.succ; // first avanza di un elemento
			p.succ.succ = null; // il successivo della (vecchia)
								// coda punta a null
		}

		return (p != null);
	}

File: src - c++/Esercizi/SpostaTestaInCoda/SpostaTestaInCoda.cpp
//Metodo privato

PRec Lista::getPuntCoda() const {
//Restituisce il puntatore alla coda della lista
  if (first) {
    PRec p = first;
    while (p->succ)
      p = p->succ;
    
    return p;
  }
  else
    return 0; //non esiste una coda se la lista è vuota
}  

//Metodo pubblico

bool Lista::SpostaTestaInCoda() {
  PRec p = getPuntCoda(); //restituisce il punt. alla coda, se c'è.
  if (p) {
    p->succ = first;
    first = first->succ;
    p->succ->succ = 0;
  }

  return (p != 0); //se p non è 0, lo spostamento è stato effettuato
}
